src/mds/ServiceDaemon.cc

```diff
@@ -0,0 +1,99 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "ServiceDaemon.h"
+#include "mgr/MgrClient.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_mds
+#undef dout_prefix
+#define dout_prefix *_dout << __func__ << ": mds.service_daemon"
+
+ServiceDaemon::ServiceDaemon(MgrClient *mgrc)
+  : mgrc(mgrc),
+    lock(ceph::make_mutex("mds::service::daemon")) {
+}
+
+ServiceDaemon::~ServiceDaemon() {
+}
+
+int ServiceDaemon::init() {
+  dout(20) << dendl;
+
+  std::lock_guard<ceph::mutex> locker(lock);
```

Comment:
`std::lock_guard` is considered obsolete. Prefer instead `std::scoped_lock`. You should also be able to get rid of the `<ceph::mutex>` due to template deduction.

Reply:
ack